packages/cli/src/modules/insights/insights.service.ts

```diff
@@ -38,6 +38,7 @@ export class InsightsService {
 		this.logger = this.logger.scoped('insights');
 	}
 
+	@OnLeaderTakeover()
```

Comment:
Since this file is imported by `FrontendService`, the decorators will run and start the pruning even when `insights` is disabled if:
* the customer runs multi main
* the leader changes

I guess we have to remove `InsightsService` from `FrontendService` before we can make this change.

Reply:
I guess we can also extract the part that the `FrontendService` needs out for now. E.g. the either:
* have the `FrontendService` use `LicenseState` directly to get the values
* extract `InsightsService.getAvailableDateRanges` into a util file with no dependencies, the `LicenseState` instance can be passed as the parameter